api/v1beta1/flowcollector_webhook.go

```diff
@@ -41,19 +42,6 @@ func (r *FlowCollector) ConvertTo(dstRaw conversion.Hub) error {
 	// Manually restore data.
 	restored := &v1beta2.FlowCollector{}
 	if ok, err := utilconversion.UnmarshalData(r, restored); err != nil || !ok {
-		// fallback on current loki config as Manual mode if metadata are not available
```

Comment:
Moving the code to `Convert_v1beta1_FlowCollectorLoki_To_v1beta2_FlowCollectorLoki` seems to work as same as current:
```
// This function need to be manually created because conversion-gen not able to create it intentionally because
// we have new defined fields in v1beta2 not in v1beta1
// nolint:golint,stylecheck,revive
func Convert_v1beta1_FlowCollectorLoki_To_v1beta2_FlowCollectorLoki(in *FlowCollectorLoki, out *v1beta2.FlowCollectorLoki, s apiconversion.Scope) error {
	// fallback on current loki config as Manual mode if metadata are not available
	out.Mode = v1beta2.LokiModeManual
	out.Manual.IngesterURL = in.URL
	out.Manual.QuerierURL = in.QuerierURL
	out.Manual.StatusURL = in.StatusURL
	out.Manual.TenantID = in.TenantID
	out.Manual.AuthToken = in.AuthToken
	if err := Convert_v1beta1_ClientTLS_To_v1beta2_ClientTLS(&in.TLS, &out.Manual.TLS, nil); err != nil {
		return fmt.Errorf("copying v1beta1.Loki.TLS into v1beta2.Loki.Manual.TLS: %w", err)
	}
	if err := Convert_v1beta1_ClientTLS_To_v1beta2_ClientTLS(&in.StatusTLS, &out.Manual.StatusTLS, nil); err != nil {
		return fmt.Errorf("copying v1beta1.Loki.StatusTLS into v1beta2.Loki.Manual.StatusTLS: %w", err)
	}
	return autoConvert_v1beta1_FlowCollectorLoki_To_v1beta2_FlowCollectorLoki(in, out, s)
}
```
I admitt it's a better place for this !

However in both case, I get the following error running `TestBeta1ConversionRoundtrip_Loki` / `TestBeta2ConversionRoundtrip_Loki` on my local env:
```
fatal error: concurrent map read and map write

goroutine 2405 [running]:
github.com/netobserv/network-observability-operator/pkg/watchers.(*Watcher).isWatched(...)
	/network-observability-operator/pkg/watchers/watcher.go:92
github.com/netobserv/network-observability-operator/pkg/watchers.(*Watcher).watch.func1({0x404e89?, 0xc000da8b60?}, {0x1fc6f00?, 0xc0010b1c20?})
	/network-observability-operator/pkg/watchers/watcher.go:77 +0x108
sigs.k8s.io/controller-runtime/pkg/handler.(*enqueueRequestsFromMapFunc).mapAndEnqueue(0x1480?, {0x1fb22d0?, 0xc000a9cc40?}, {0x1fba648, 0xc000940420}, {0x1fc6f00?, 0xc0010b1c20?}, 0x0?)
	/network-observability-operator/vendor/sigs.k8s.io/controller-runtime/pkg/handler/enqueue_mapped.go:81 +0x70
sigs.k8s.io/controller-runtime/pkg/handler.(*enqueueRequestsFromMapFunc).Create(0x1fb22d0?, {0x1fb22d0, 0xc000a9cc40}, {{0x1fc6f00?, 0xc0010b1c20?}}, {0x1fba648, 0xc000940420})
	/network-observability-operator/vendor/sigs.k8s.io/controller-runtime/pkg/handler/enqueue_mapped.go:58 +0xe8
sigs.k8s.io/controller-runtime/pkg/internal/source.(*EventHandler).OnAdd(0xc0016ea190, {0x1c47820?, 0xc0010b1c20})
	/network-observability-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/source/event_handler.go:88 +0x243
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/network-observability-operator/vendor/k8s.io/client-go/tools/cache/controller.go:239
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/network-observability-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:974 +0x148
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
	/network-observability-operator/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc001c22738?, {0x1f98680, 0xc0006f5d10}, 0x1, 0xc000d21da0)
	/network-observability-operator/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/network-observability-operator/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/network-observability-operator/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161
k8s.io/client-go/tools/cache.(*processorListener).run(0xc001601cb0?)
	/network-observability-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:968 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/network-observability-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:72 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/network-observability-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:70 +0x85
```

Reply:
marking as resolved, as this was another bug (fixed)